src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java

```diff
@@ -0,0 +1,184 @@
+        assertThat(this.database.getUserByLogin(STUDENT_NAME)).as("User shall exist").isNotNull();
+
+        // Change Password
+        String password = "test123";
+        User student = userRepository.findUserWithGroupsAndAuthoritiesByLogin(STUDENT_NAME).get();
+        student.setPassword(passwordService.encryptPassword(password));
+        userRepository.saveAndFlush(student);
+
+        // Try to login ..
+        TestSecurityContextHolder.clearContext();
+        LoginVM user = new LoginVM();
+        user.setUsername(STUDENT_NAME);
+        user.setPassword(password);
+        user.setRememberMe(true);
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.add("User-Agent", "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36");
+
+        // Test whether authorize generates a valid token
+        UserJWTController.JWTToken result = request.postWithResponseBody("/api/authenticate", user, UserJWTController.JWTToken.class, HttpStatus.OK, httpHeaders);
+        assertThat(this.tokenProvider.validateToken(result.getIdToken())).as("JWT Token is Valid").isTrue();
+
+        // Check SAML Login afterwards ..
+
+        TestSecurityContextHolder.clearContext();
+        // Mock existing SAML2 Auth
+        authentication = new Saml2Authentication(createPrincipal(STUDENT_NAME), "Secret Credentials", null);
+        TestSecurityContextHolder.setAuthentication(authentication);
+        // Test whether authorizeSAML2 generates a valid token
+        result = request.postWithResponseBody("/api/saml2", Boolean.FALSE, UserJWTController.JWTToken.class, HttpStatus.OK);
+        assertThat(this.tokenProvider.validateToken(result.getIdToken())).as("JWT Token is Valid").isTrue();
+    }
+
+    /**
+     * This tests checks whether the access to the system is restricted if the login is not present.
+     *
+     * @throws Exception if something went wrong
+     */
+    @Test
+    public void testInvalidAuthenticationSaml2Login() throws Exception {
+        assertThatThrownBy(() -> this.database.getUserByLogin(STUDENT_NAME)).isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Provided login student1 does not exist in database");
+
+        // Test whether authorizeSAML2 generates a no token
+        request.post("/api/saml2", Boolean.FALSE, HttpStatus.UNAUTHORIZED);
+
+        assertThatThrownBy(() -> this.database.getUserByLogin(STUDENT_NAME)).isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Provided login student1 does not exist in database");
+    }
+
+    @AfterEach
+    public void clearAuth() {
+        TestSecurityContextHolder.clearContext();
+        this.database.resetDatabase();
+    }
+
+    private Saml2AuthenticatedPrincipal createPrincipal(String username) {
+        Saml2AuthenticatedPrincipal principal = new DefaultSaml2AuthenticatedPrincipal(username, new HashMap<>() {
+
+            {
```

Comment:
I guess you could use something like Map.of(...) here to avoid the Static Code Analysis Warning here

Reply:
I'll take a look

Reply:
Should be resolved now. I'll test all changes on our test servers.